node/core/candidate-validation/src/lib.rs

```diff
@@ -467,11 +467,6 @@ where
 	.await;
 
 	if let Ok(ValidationResult::Valid(ref outputs, _)) = validation_result {
-		// If validation produces new commitments we consider the candidate invalid.
```

Comment:
I might be missing something, but I don't find the commitments hash check at upper level -  the `candidate-backing` subsystem.

Reply:
It's the same check as in approval-voting - `validate_candidate_exhaustive`

Reply:
In approval voting I can see it being called from https://github.com/paritytech/polkadot/blob/1eb107d26dd7ea692e61d157d9c4c5ec78d1b0ac/node/core/approval-voting/src/lib.rs#L2395

but from backing we will call `validate_from_chain_state` via https://github.com/paritytech/polkadot/blob/702d4fc7d502ae7cbfa0a4011101129c5bb5fff6/node/core/backing/src/lib.rs#L734

and I don't see any hash check, it just matches on `InvalidCandidate::CommitmentsHashMismatch` which is no longer be returned with this change.

Reply:
I meant that `validate_from_chain_state` is itself calling `validate_candidate_exhaustive` (scroll above), which does this check.

Reply:
Ah, indeed it does and makes total sense